src/commonMain/kotlin/app/cash/turbine/flow.kt

```diff
@@ -121,10 +102,17 @@ private fun <T> Flow<T>.collectTurbineIn(scope: CoroutineScope): Turbine<T> {
     ?: Unconfined
 
   val job = scope.launch(unconfined, start = UNDISPATCHED) {
-    channel = collectIntoChannel(this)
+    val block: suspend CoroutineScope.() -> Unit = {
+      channel = collectIntoChannel(this)
+    }
+    if (timeout != null) {
+      withTurbineTimeout(timeout, block)
+    } else {
+      block()
+    }
```

Comment:
Why do this instead of passing the timeout into `ChannelTurbine`?

Reply:
1. Passing the timeout into `ChannelTurbine` wraps every single `receive()` in a `withContext` call, while this only requires one
2. Doing this means that the timeout is also applied to any other Turbines run within this block.

Reply:
I don't think either of those things apply here. The `block` is only a call to `flow.collect { channel.trySend(it) }` and never runs user code (unless they're using Turbine inside `flow { }`). This coroutine's context isn't shared by the caller interacting with the `Turbine` instance.

I'll write a quick test for it.

Reply:
Oof, you're right. We probably don't even need to be hooking up the timeout at all here: the timeout isn't even accessed in this context.

Reply:
Issue addressed; new tests in `FlowTest` and `FlowInScopeTest`

Reply:
We probably just wrote the same tests. Signing off for a few hours.